core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -597,6 +586,9 @@ private[spark] class BlockManager(
 
   /**
    * Get a block from the block manager (either local or remote).
+   *
+   * This acquires a read lock on the block if the block was stored locally and does not acquire
+   * any locks if the block was fetched from a remote block manager.
```

Comment:
we should expand on this comment that the caller may check whether the `readMethod` of the `BlockResult` is network, and if so, do not release the lock (i.e. what we do in `CacheManager`)